src/swarm/neo/client/mixins/BatchRequestCore.d

```diff
@@ -0,0 +1,257 @@
+    {
+        import ocean.core.Enforce;
+        import swarm.neo.client.mixins.RequestCore : ControllerBase;
+
+        /***********************************************************************
+
+            Base mixin.
+
+        ***********************************************************************/
+
+        mixin ControllerBase;
+
+        /***********************************************************************
+
+            Tells the nodes to stop sending data to this request.
+
+            Returns:
+                true because this controller function can always be used
+
+        ***********************************************************************/
+
+        public bool suspend ( )
+        {
+            Request.getContext(this.request_controller.context_blob)
+                .shared_working.suspendable_control.suspended = true;
+            return true;
+        }
+
+        /***********************************************************************
+
+            Tells the nodes to resume sending data to this request.
+
+            Returns:
+                true because this controller function can always be used
+
+        ***********************************************************************/
+
+        public bool resume ( )
+        {
+            Request.getContext(this.request_controller.context_blob)
+                .shared_working.suspendable_control.suspended = false;
+            this.request_controller.resumeSuspendedHandlers(
+                BatchRequestSharedWorkingData.Signal.Resume);
+            return true;
+        }
+
+        /***********************************************************************
+
+            Tells the nodes to cleanly end the request.
+
+            Returns:
+                true because this controller function can always be used
+
+        ***********************************************************************/
+
+        public bool stop ( )
+        {
+            with (Request.getContext(this.request_controller.context_blob).shared_working.suspendable_control)
+            {
+                if (!stopped)
+                {
+                    stopped = true;
+                    this.request_controller.resumeSuspendedHandlers(
+                        BatchRequestSharedWorkingData.Signal.Stop);
+                }
+            }
+            return true;
+        }
+    }
+}
+
+///
+unittest
+{
+    // Example request struct with RequestCore and BatchController mixed in
+    struct ExampleRequest
+    {
+        mixin ExampleRequestCore!();
+
+        // Required by BatchController
+        interface IController
+        {
+            bool suspend ( );
+            bool resume ( );
+            bool stop ( );
+        }
+
+        mixin BatchController!(ExampleRequest, IController);
+    }
+}
+
+/*******************************************************************************
+
+    Data required by batchRequestConnector, BatchRequestInitialiser,
```

Comment:
`batchRequestConnector` and `BatchRequestInitialiser` are mentioned here but do not appear in this module. I guess this comment is cut & pasted from the old suspendable core stuff and these things are no longer needed / relevant.

Reply:
Yes, I removed most mentions from the comments but didn't catch all of them.